### ISSUE TITLE
Form retaining responses

### ISSUE THREAD

**Issue by antenehtassalif** [odk-x/tool-suite-X#159]
When you go back and change the response of a question, all dependent values need to be cleared. Currently, the form retains values even after the response to a parent question is changed. It is possible to reset values through assign but this won't be safe when we have a large survey; plus, it also contributes to the size and complexity of forms.

**Comment by elmps2018** [comment]
This may not be true for all use cases. For example, in longitudinal data collection, it may be that there are sections (such as demographic information) you want to collect on the first visit, and retain for future visits, but not re-ask. So if there's a "which is the latest visit?" question and you update to the second visit, you don't want demographics to get deleted.

**Comment by e2themillions** [comment]
I think the danger is that you could wind up with an answer to a question that you didn't ask.

E.g. if you have 2 questions, where no. 2 is only asked when no. 1 has answer X. 
If you answer "X" to no. 1 and then answer "something" to question no. 2 and goes back and change the answer to question no. 1 to Y (which will cause question no. 2 to be skipped) - the value of question no. 2 will still be "something".

I have worked around this using calculate fields, that basically says "if number 2 is visible, take the answer of number 2, otherwise be blank". But that does indeed contribute to the size and complexity of the form.

I think this this a bug?

**Comment by e2themillions** [comment]
As a side note - in ODK Collect there is the option of "removing" an answer entirely by long-tapping. I think that is basically because of this?

**Comment by elmps2018** [comment]
I am not sure this is a bug though in longitudinal data collection. I understand the example and cases where you end up with "extra" data. However, say you have a form that collects data longitudinally (and there are many of these). If there is a skip pattern based on the visit, it might go like this:
Choose visit 1: answer demographic section and health section
Choose visit 2: skip demographic section and do only health section

You would not want the demographic section to get deleted on the second visit. So this would create a problem in that use case. 

It might need to be an option, rather than a requirement, to "empty out" questions that get skipped.

**Comment by e2themillions** [comment]
I agree that we must be careful not to create new inconveniences :-) 
I am not sure I fully understand the longitudinally case; Wouldn't two visits be recorded in two separate questionnaire instances?
Perhaps we could discuss this topic at the next community call? :-)

**Comment by wbrunette** [comment]
A proposal to resolve the issue. Create a new feature with a column in the XLSX survey sheet that will let the system know to clear this value if the conditionals do not match. The idea is when the form is being converted to formDef.json is to automatically create clearing formulas based on the if conditionals. The idea is pre-process and throw compile errors if the situation is complicated by goto's, etc.

**Comment by antenehtassalif** [comment]
nice proposal @wbrunette.
